### PR TITLE
Feature/upload-download

### PR DESCRIPTION
**Pull request by dinhl**

### CODE REVIEW

File: dep/proto/hwsc-file-svc.proto [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
-option go_package="hwscFileSvcPb";
+option go_package="hwscFileTransactionSvcPb";
 
-service FileService{
-    rpc GetStatus(FileRequest) returns (FileResponse) {}
-}
+// Service: define the methods that gRPC server can expose to the client
+service FileTransactionService{
+    rpc GetStatus(FileTransactionRequest) returns (FileTransactionResponse) {}
+    rpc upload(stream Segment)returns(Reply){}
```
Comment (faraonc) [inline dep/proto/hwsc-file-svc.proto]: rename as UploadFile

File: dep/proto/hwsc-file-svc.proto [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
     FAIL = 3;
 }
 
-message FileRequest{
+message FileTransactionRequest{
     string message = 1;
 }
 
-message FileResponse{
+message FileTransactionResponse{
```
[the remark this answers is not in the document]
Comment (faraonc) [reply dep/proto/hwsc-file-svc.proto]: also need ``string url`` field

**Comment by faraonc** [comment]
added @kimlisa @wai-swk we need a strong proto for scalability

**Comment by faraonc** [comment]
rename file to hwsc-file-transaction-svc.proto

File: dep/proto/hwsc-file-svc.proto [hunk 3]
```diff
@@ -1,24 +1,41 @@
 syntax="proto3";
 
-package hwscFileSvc;
+package hwscFileTransactionSvc;
 
-option go_package="hwscFileSvcPb";
+option go_package="hwscFileTransactionSvcPb";
```
Comment (kimlisa) [inline dep/proto/hwsc-file-svc.proto]: delete line, option go_package is only useful when using go lang

**Comment by faraonc** [comment]
this is closed in favor of the https://github.com/faraonc/hwsc-api-blocks/pull/42

**Pull request closed without merging**